agents/opencode_launcher.py

```
@@ -0,0 +1,150 @@
+        """Start OpenCode server and wait for it to be ready.
+
+        Returns:
+            base_url: The URL of the OpenCode server.
+        """
+        if self._process is not None:
+            return self.base_url
+
+        logger.info(f"Starting OpenCode server on {self.base_url}")
+        cmd = ["opencode", "serve", "--port", str(self.port), "--hostname", self.hostname]
+
+        self._process = subprocess.Popen(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)